controllers/datadogagent/clusteragent.go

```diff
@@ -763,7 +766,10 @@ func getClusterAgentMetricsProviderPort(config datadoghqv1alpha1.ClusterAgentCon
 }
 
 func getAdmissionControllerServiceName(dda *datadoghqv1alpha1.DatadogAgent) string {
-	if isClusterAgentEnabled(dda.Spec.ClusterAgent) && dda.Spec.ClusterAgent.Config.AdmissionController != nil && dda.Spec.ClusterAgent.Config.AdmissionController.ServiceName != nil {
+	if isClusterAgentEnabled(dda.Spec.ClusterAgent) &&
+		dda.Spec.ClusterAgent.Config != nil &&
```

Comment:
why is the config necessary for the admission controller?

Reply:
because the function takes value in `dda.Spec.ClusterAgent.Config.AdmissionController`.
due to the bug in the `DefaultClusterAgent()` when `if IsEqualStruct(*clusterAgent, DatadogAgentSpecClusterAgentSpec{}) ` was `true`, the `dda.Spec.ClusterAgent.Config` was `nil`. 

I think it is ok to keep this change, because it is always better to check point value before using it.